File: WebStoreAngular/src/app/models/orders/order-response.ts
import { DeliveryResponse } from './delivery-response';
import { ShipmentResponse } from './shipment-response';

export interface OrderResponse {
  id: string;
  nameUser: string;
  totalPrice: number;
  delivery: DeliveryResponse;
  dateOfCreation: Date;
  status: string;
  shipments: ShipmentResponse[];
}

File: WebStoreSpring/src/main/java/com/example/portfolio/webstorespring/models/dto/orders/response/OrderResponse.java
package com.example.portfolio.webstorespring.models.dto.orders.response;

import com.example.portfolio.webstorespring.enums.OrderStatus;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

public record OrderResponse(Long id,
                            String nameUser,
                            BigDecimal totalPrice,
                            @JsonProperty(value = "delivery")
                            DeliveryResponse deliveryResponse,
                            LocalDateTime createdAt,
                            OrderStatus status,
                            @JsonProperty(value = "shipments")
                            List<ShipmentResponse> shipmentResponses
) {
}
